question-service/routes/routes.js

```diff
@@ -38,6 +38,74 @@ router.get('/questions/:id', async (req, res) => {
     }
 })
 
+//Get random question by complexity and category
+// http://localhost:3000/api/random?complexity=<complexity>&categories=<categories>
+router.get('/random', async (req, res) => {
+    const { complexity, categories } = req.query;
+    
+    // Filter questions based on complexity and category
+    // can be empty for both
+    // params are case sensitive
+    const filter = {};
+    if (complexity) {
+        filter.complexity = complexity;
+    }
+    if (categories) {
+        filter.categories = categories;
+    }
+
+    try {
+      // Use your Mongoose model to query the database
+      const filteredQuestions = await Question.find(filter);
+
+      // Check if any questions match the criteria
+      if (filteredQuestions.length === 0) {
+        return res
+          .status(404)
+          .json({ message: "No questions found with the specified criteria" });
+      }
+
+      // Select a random question from the filtered list
+      const randomIndex = Math.floor(Math.random() * filteredQuestions.length);
+      const randomQuestion = filteredQuestions[randomIndex];
+
+      res.json(randomQuestion);
+    } catch (error) {
+        res.status(500).json({ error: "Error fetching random question" });
+    }
+});
+
+// filter by choice
+router.get("/filter", async (req, res) => {
+  const { categories, complexity } = req.query;
+
+  // can be empty for any
+  // params are case sensitive
+    const filter = {};
+    if (categories) {
+        filter.categories = categories;
+    }
+    if (complexity) {
+        filter.complexity = complexity;
+    }
+
+  try {
+    // Use your Mongoose model to query the database
+    const filteredQuestions = await Question.find(filter);
+
+    // Check if any questions match the criteria
+    if (filteredQuestions.length === 0) {
+      return res
+        .status(404)
```

Comment:
I think it would be better to return a 200 OK response code since the request is actually successful, even if it returns an empty array. The frontend can just check the length of the returned data array and display an appropriate message.